src/app/archive/archive_lib/metrics.ml

```diff
@@ -0,0 +1,94 @@
+open Core_kernel
+open Async
+
+let time ~label f =
+  let start = Time.now () in
+  let%map x = f () in
+  let stop = Time.now () in
+  [%log' info (Logger.create ())]
+    "%s took %s" label
+    (Time.Span.to_string_hum (Time.diff stop start)) ;
+  x
+
+module Max_block_height = struct
+  let query =
+    Caqti_request.find Caqti_type.unit Caqti_type.int
+      "SELECT max(height) FROM blocks"
+
+  let update (module Conn : Caqti_async.CONNECTION) metric_server =
+    time ~label:"max_block_height" (fun () ->
+        let open Deferred.Result.Let_syntax in
+        let%map max_height = Conn.find query () in
+        Mina_metrics.(
+          Gauge.set
+            (Archive.max_block_height metric_server)
+            (Float.of_int max_height)) )
+end
+
+module Missing_blocks = struct
+  (*A block is missing if there is no entry for a specific height. However, if there is an entry then it doesn't necessarily mean that it is part of the main chain. Unparented_blocks will show value > 1 in that case. Look for the last 2000 blocks*)
+  let query =
+    Caqti_request.find Caqti_type.unit Caqti_type.int
+      (Core_kernel.sprintf
+         {sql| 
+        SELECT count( * )
+        FROM (SELECT h::int FROM generate_series(GREATEST(1, (select max(height) from blocks)-2000) , (select max(height) from blocks)) h
```

Comment:
should the number of blocks be a parameter somewhere, instead of hard-coded here?